experimental/generator-dotnet-yeoman/generators/app/index.js

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+const Generator = require('yeoman-generator');
+const path = require('path');
+const { v4: uuidv4 } = require('uuid');
+
+module.exports = class extends Generator {
+    constructor(args, opts) {
+        super(args, opts);
+
+        this.argument('botName', {
+            type: String,
+            required: true
+        });
+        this.argument('integration', {
+            type: String,
+            required: true
+        });
+    }
+
+    // 1. initializing - Your initialization methods (checking current project state, getting configs, etc)
+    // 2. prompting - Where you prompt users for options (where you’d call this.prompt())
+    // 3. configuring - Saving configurations and configure the project (creating .editorconfig files and other metadata files)
+    // 4. default - If the method name doesn’t match a priority, it will be pushed to this group.
+    // 5. writing - Where you write the generator specific files (routes, controllers, etc)
+    // 6. conflicts - Where conflicts are handled (used internally)
+    // 7. install - Where installations are run (npm, bower)
+    // 8. end - Called last, cleanup, say good bye, etc
+
+    writing() {
+        this._copyBotProject(this.options.integration);
+        this._copySolutionFile();
+    }
+
+    _copyBotProject(integration) {
+        if (this.options.integration == "webapp" || this.options.integration == "functions") {
```

Comment:
This check should probably be a validation, otherwise you could enter a random value and end up not copying something. Probably makes sense to do that in the constructor.